clients/vault/src/oracle/agent.rs

```diff
@@ -15,16 +18,40 @@ use crate::oracle::{
 	collector::ScpMessageCollector, errors::Error, types::StellarMessageSender, AddTxSet, Proof,
 };
 use wallet::Slot;
+use crate::tokio_spawn;
 
 pub struct OracleAgent {
-	collector: Arc<RwLock<ScpMessageCollector>>,
+	pub collector: Arc<RwLock<ScpMessageCollector>>,
 	pub is_public_network: bool,
 	/// sends message directly to Stellar Node
 	message_sender: Option<StellarMessageSender>,
 	/// sends an entire Vault shutdown
-	shutdown_sender: ShutdownSender,
-	/// sends a 'stop' signal to `StellarOverlayConnection` poll
-	overlay_conn_end_signal: mpsc::Sender<()>,
```

Comment:
We do not need this signal anymore; OverlayConnection already implements `Drop` trait.